### ISSUE TITLE
ReactiveViewHost KeyNotFoundException after update to version 7.1

### ISSUE THREAD

**Issue by bidy** [reactiveui/ReactiveUI#1227]
I'm doing this in the Xamarin.Android Activity:
`var adapter = new ReactiveListAdapter<TreeItemViewModel>(ViewModel.TreeItemViewModels, (viewModel, parent) => new ImageItemView(viewModel, this, parent));`

the ImageItemView is like this:

`public class ImageItemView : ReactiveViewHost<TreeItemViewModel>
    {
        public ImageItemView(TreeItemViewModel viewModel, Context ctx, ViewGroup parent)
            : base(ctx, Resource.Layout.drawer_list_item, parent)
        {
            ViewModel = viewModel;

            this.OneWayBind(ViewModel, vm => vm.Title, v => v.Text1.Text);
            this.WhenAnyValue(vm => vm.ViewModel.Icon)
                .Subscribe(x => ImgIcon.SetImageDrawable(x.ToNative()));
        }

        public ImageView ImgIcon { get; private set; }
        public TextView Text1 { get; private set; }
    }`

This worked well in ReactiveUI Version 6. Now with version 7.1 I get this exception:
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

The exception occurs when calling the base constructor of ImageItemView.

The views / Layouts / xaml haven't been changed and worked with the previous version.

**Comment by ghuntley** [comment]
Can you please provide the entire stacktrace?

**Comment by bidy** [comment]
Call stack:
 	0xFFFFFFFFFFFFFFFF in System.Diagnostics.Debugger.Mono_UnhandledException_internal	C#
 	0x1 in System.Diagnostics.Debugger.Mono_UnhandledException at /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/corlib/System.Diagnostics/Debugger.cs:122,-1	C#
 	0x33 in object.b972e2c9-785a-4893-8f94-61ba0a6f3aa9	C#
 	0x48 in ReactiveUI.ControlFetcherMixin.	C#
 	0x42 in System.Collections.Generic.List<System.Reflection.PropertyInfo>.ForEach at /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/collections/generic/list.cs:564,-1	C#
 	0x54 in ReactiveUI.ControlFetcherMixin.WireUpControls	C#
 	0x22 in ReactiveUI.LayoutViewHost..ctor	C#
 	0x8 in ReactiveUI.ReactiveViewHost<AndroidUi.TreeItemViewModel>..ctor	C#
>	0xA in AndroidUi.Droid.ImageItemView..ctor at C:\Projects\XamarinPrototype\AndroidUi\AndroidUi\AndroidUi.Droid\ImageItemView.cs:14,15	C#
 	0x3 in AndroidUi.Droid.MainActivity.<OnCreate>b__6_1 at C:\Projects\XamarinPrototype\AndroidUi\AndroidUi\AndroidUi.Droid\MainActivity.cs:48,131	C#
 	0x1A in ReactiveUI.ReactiveListAdapter<AndroidUi.TreeItemViewModel>.GetView	C#
 	0x1E in Android.Widget.BaseAdapter.n_GetView_ILandroid_view_View_Landroid_view_ViewGroup_ at /Users/builder/data/lanes/3511/501e63ce/source/monodroid/src/Mono.Android/platforms/android-24/src/generated/Android.Widget.BaseAdapter.cs:443,4	C#
 	0x23 in object.b972e2c9-785a-4893-8f94-61ba0a6f3aa9	C#

**Comment by bidy** [comment]
Am I the only one with this issue?

**Comment by makon** [comment]
No you are not the only one with this issue.

Unfortunately this issues comes from #1217 where WireUpControls also tries to find properties of type View and not only from subtypes of View. This leads to the issue that the property "View" in LayoutViewHost cannot be found and a KeyNotFoundException occurs.

```
public abstract class LayoutViewHost : ILayoutViewHost, IEnableLogger
{
    View view;
    public View View {
        get { return view; }
        set {
            if (view == value) return;
            view = value;
            view.SetTag(ViewMixins.viewHostTag, this.ToJavaObject());
        }
    }
```

But this issue isn't only related to ReactiveViewHost. Through this change WireUpControls is broken because many base classes of android have properties of type View which it will not find. (e.g. Activity, Fragment etc.) :-(

**Comment by Qonstrukt** [comment]
This is fixed in 7.2.0 with #1250. Thanks @bidy for reporting, and @makon for thinking on a solution.
